mbed_tools_ci/utils/package_helpers.py

```diff
@@ -0,0 +1,178 @@
+        return self._data.get('Name', UNKNOWN)
+
+    @property
+    def version(self) -> str:
+        """Gets package's version."""
+        return self._data.get('Version', UNKNOWN)
+
+    @property
+    def author(self) -> str:
+        """Gets package's author."""
+        return self._data.get('Author', UNKNOWN)
+
+    @property
+    def author_email(self) -> str:
+        """Gets package's author email."""
+        return self._data.get('Author-email', UNKNOWN)
+
+    @property
+    def licence(self) -> str:
+        """Gets package's licence."""
+        return self._data.get('License', UNKNOWN)
+
+    @property
+    def description(self) -> str:
+        """Gets package's licence."""
+        return self._data.get('Summary', UNKNOWN)
+
+    @property
+    def url(self) -> str:
+        """Gets package's URL."""
+        home_page = str(self._data.get('Home-page'))
+        if home_page:
+            return home_page
+        url = str(self._data.get('Project-URL'))
+        if url:
+            return url.split(',')[1].strip()
+        return UNKNOWN
+
+    def __str__(self) -> str:
+        """String representation."""
+        relevant_data = [
+            f'{getter}: {getattr(self, getter, None)}' for getter
+            in dir(self) if not getter.startswith("_") and getter
+        ]
+        return f'{", ".join(relevant_data)}'
+
+
+class ProjectMetadata:
+    """Metadata for a project."""
+
+    def __init__(self, package_name: str) -> None:
+        """Constructor."""
+        self._package_metadata: PackageMetadata = PackageMetadata(dict())
+        self._dependency_packages_metadata: List[PackageMetadata] = []
+        self._package_name: str = package_name
+
+    @property
+    def dependencies_metadata(self) -> List[PackageMetadata]:
+        """Gets all package's dependencies metadata."""
+        return self._dependency_packages_metadata
+
+    @property
+    def project_metadata(self) -> PackageMetadata:
+        """Gets project metadata."""
+        return self._package_metadata
+
+    @property
+    def package_name(self) -> str:
+        """Gets project's package name."""
+        return self._package_name
+
+    def __str__(self) -> str:
+        """String representation."""
+        dependencies_str = " | ".join(
+            [str(t) for t in self._dependency_packages_metadata])
+        project_str = f'Project [{self.package_name}]'
+        metadata_str = str(self._package_metadata)
+        other_str = f'Dependencies: [{dependencies_str}]'
+
+        return f'{project_str}: {metadata_str};  {other_str}'
+
+
+class ProjectMetadataParser:
+    """Parser of python package metadata."""
+    ENTRY_PATTERN = r'^([^:]*):(.*)$'
+
+    def __init__(self, package_name: str) -> None:
+        """Constructor."""
+        self._project_metadata: Optional[ProjectMetadata] = None
+        self._package_name = package_name
+
+    @staticmethod
+    def _get_metadata_lines(package: Any) -> Any:
+        try:
```

Comment:
Do we really want to accept and return `Any` type and catch every possible `Exception` to return an empty list?

Reply:
Depending on the package, there may be a METADATA file or a PKG-INFO. We need to try both to get the metadata lines and the `get_metadata_lines` raises an exception if the file does not exist. So yes, we need to try to catch all exceptions. The idea is if everything goes well, we get package metadata, otherwise we don't but we don't stop the process

Reply:
I think it would be worth adding a comment explaining this in the code.